### PR TITLE
Merge with core 

### PR DESCRIPTION
**Pull request by AnieeG**
## Solution

### CODE REVIEW

File: core/chains/evm/gas/block_history_estimator_test.go [hunk 1]
```diff
@@ -1340,7 +1341,7 @@ func TestBlockHistoryEstimator_IsUsable(t *testing.T) {
 	t.Run("returns false if transaction is of type 0x16 only on WeMix", func(t *testing.T) {
 		cfg.ChainTypeF = "wemix"
 		tx := evmtypes.Transaction{Type: 0x16, GasPrice: assets.NewWeiI(10), GasLimit: 42, Hash: utils.NewHash()}
-		assert.Equal(t, false, bhe.IsUsable(tx, block, cfg.ChainType(), geCfg.PriceMin(), logger.TestLogger(t)))
```
Comment (RensR) [inline core/chains/evm/gas/block_history_estimator_test.go:1335]: Why are changes needed to a non-CCIP component?

File: core/services/chainlink/application.go [hunk 2]
```diff
@@ -241,7 +241,9 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 	}
 
 	// pool must be started before all relayers and stopped after them
-	srvcs = append(srvcs, opts.MercuryPool)
+	if opts.MercuryPool != nil {
```
Comment (RensR) [inline core/services/chainlink/application.go:244]: Why is this new code? Whats the reasoning here

File: core/services/fluxmonitorv2/delegate.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ func (d *Delegate) BeforeJobDeleted(spec job.Job)                {}
 func (d *Delegate) OnDeleteJob(spec job.Job, q pg.Queryer) error { return nil }
 
 // ServicesForSpec returns the flux monitor service for the job spec
-func (d *Delegate) ServicesForSpec(jb job.Job) (services []job.ServiceCtx, err error) {
```
Comment (RensR) [inline core/services/fluxmonitorv2/delegate.go]: Why was this needed?

File: core/services/ocr2/plugins/ccip/internal/ccipdata/onramp_reader_test.go [hunk 4]
```diff
@@ -1,12 +1,14 @@
 package ccipdata_test
 
 import (
-	evmclientmocks "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client/mocks"
-	"github.com/stretchr/testify/mock"
 	"math/big"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/mock"
```
Comment (RensR) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/onramp_reader_test.go]: nit: this import is still wrong, there should be no newlines in between so it auto formats correctly. Same for a few others here

File: core/services/ocr2/plugins/ccip/testhelpers/integration/jobspec.go [hunk 5]
```diff
@@ -1,21 +1,145 @@
...
 
 	"github.com/smartcontractkit/chainlink/v2/core/services/job"
 	"github.com/smartcontractkit/chainlink/v2/core/services/relay"
 	"github.com/smartcontractkit/chainlink/v2/core/store/models"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
 )
 
+// OCR2TaskJobSpec represents an OCR2 job that is given to other nodes, meant to communicate with the bootstrap node,
+// and provide their answers
+type OCR2TaskJobSpec struct {
```
Comment (RensR) [inline core/services/ocr2/plugins/ccip/testhelpers/integration/jobspec.go:22]: Massive block of code added, could someone add an explanation why?

File: docs/CONFIG.md [hunk 6]
```diff
@@ -2906,7 +2904,7 @@ AutoCreateKey = true
 BlockBackfillDepth = 10
 BlockBackfillSkip = false
 ChainType = 'kroma'
-FinalityDepth = 40
```
Comment (RensR) [inline docs/CONFIG.md]: Feels like we should be in sync with main here. If 400 is the value we want, we need to also change it in main?

File: docs/CONFIG.md [hunk 7]
```diff
@@ -3628,164 +3626,6 @@ GasLimit = 5300000
 
 </p></details>
 
-<details><summary>WeMix Mainnet (1111)</summary><p>
```
Comment (RensR) [inline docs/CONFIG.md]: ?

File: docs/CONFIG.md [hunk 8]
```diff
@@ -5567,7 +5332,7 @@ GasLimit = 5300000
 AutoCreateKey = true
 BlockBackfillDepth = 10
 BlockBackfillSkip = false
-FinalityDepth = 5
```
Comment (RensR) [inline docs/CONFIG.md]: same as above

File: integration-tests/ccip-tests/testconfig/global.go [hunk 9]
```diff
@@ -7,9 +7,13 @@ import (
 	"fmt"
 
 	"github.com/AlekSi/pointer"
+
```
Comment (RensR) [inline integration-tests/ccip-tests/testconfig/global.go]: What happened here?

File: core/services/chainlink/application.go (hunk 2 above)
Comment (makramkd) [reply core/services/chainlink/application.go:244]: It was panicking down the line because `opts.MercuryPool` is `nil`. This seems like a bug in core actually, since I wouldn't expect every app to pass in a correctly constructed `MercuryPool` object.

File: integration-tests/load/automationv2_1/gun.go [hunk 10]
```diff
@@ -28,16 +28,16 @@ func NewLogTriggerUser(
 	}
 }
 
-func (m *LogTriggerGun) Call(_ *wasp.Generator) *wasp.CallResult {
```
Comment (RensR) [inline integration-tests/load/automationv2_1/gun.go]: Feels like this should not change in ccip? Is this intentional?

File: core/services/ocr2/plugins/ccip/testhelpers/integration/jobspec.go (hunk 5 above)
Comment (makramkd) [reply core/services/ocr2/plugins/ccip/testhelpers/integration/jobspec.go:22]: I think @AnieeG added this to fix the dependency of core on integration-tests in the CCIP fork, but will let her chime in in case I'm wrong.

File: core/chains/evm/gas/block_history_estimator_test.go (hunk 1 above)
Comment (makramkd) [reply core/chains/evm/gas/block_history_estimator_test.go:1335]: Not sure why its showing up in the diff like this but this the line you see in core:

```
assert.Equal(t, false, bhe.IsUsable(tx, block, cfg.ChainType(), geCfg.PriceMin(), logger.Test(t)))
```

i.e instead of `logger.TestLogger(t)` you have `logger.Test(t)`.

Comment (makramkd) [reply core/chains/evm/gas/block_history_estimator_test.go:1335]: This probably came up as a conflict but was fixed in subsequent non-merge commits, but not sure.

File: core/services/ocr2/plugins/ccip/testhelpers/integration/jobspec.go (hunk 5 above)
Comment (AnieeG) [reply core/services/ocr2/plugins/ccip/testhelpers/integration/jobspec.go:22]: Right. It was either this or fixing the k8 dependency hell that integration-tests brought in.
I have an action item noted from this - We will need to make [integration-tests/client](https://github.com/smartcontractkit/chainlink/tree/develop/integration-tests/client) independent of rest of the integration-tests. We can then import only the client and get rid of all these duplication of codes.

File: integration-tests/ccip-tests/testconfig/global.go (hunk 9 above)
Comment (AnieeG) [reply integration-tests/ccip-tests/testconfig/global.go]: With this PR integration-tests workflow introduced lint checks. All `integration-tests/ccip-tests` changes are to fix lint erros.

File: core/services/chainlink/application.go (hunk 2 above)
Comment (RensR) [reply core/services/chainlink/application.go:244]: Probably want to raise that with core

File: core/services/fluxmonitorv2/delegate.go (hunk 3 above)
Comment (makramkd) [reply core/services/fluxmonitorv2/delegate.go]: Good catch, this seems incorrect.

File: core/services/keeper/delegate.go [hunk 11]
```diff
@@ -55,7 +55,7 @@ func (d *Delegate) BeforeJobDeleted(spec job.Job)                {}
 func (d *Delegate) OnDeleteJob(spec job.Job, q pg.Queryer) error { return nil }
 
 // ServicesForSpec satisfies the job.Delegate interface.
-func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.ServiceCtx, err error) {
+func (d *Delegate) ServicesForSpec(spec job.Job, opts ...pg.QOpt) (services []job.ServiceCtx, err error) {
```
Comment (makramkd) [inline core/services/keeper/delegate.go:58]: Also incorrect.

File: integration-tests/load/automationv2_1/gun.go (hunk 10 above)
Comment (AnieeG) [reply integration-tests/load/automationv2_1/gun.go]: Yes, intentional. I had to update wasp dependency to get rid of `go.mod` conflicts and that called for these renaming. [core Pr with similar changes](https://github.com/smartcontractkit/chainlink/pull/11590)

File: core/services/fluxmonitorv2/delegate.go (hunk 3 above)
Comment (makramkd) [reply core/services/fluxmonitorv2/delegate.go]: Seems like these have been removed in core but we still need them on CCIP.

File: docs/CONFIG.md (hunk 6 above)
Comment (makramkd) [reply docs/CONFIG.md]: That sounds good to me, can create a ticket.

File: docs/CONFIG.md (hunk 7 above)
Comment (makramkd) [reply docs/CONFIG.md]: I think this may be due to some alphabetical order changes, this is still in the file, just in another location. See https://raw.githubusercontent.com/smartcontractkit/ccip/147721d5e1490d1cde18454da026f7a8804618f8/docs/CONFIG.md and CTRL+F for "WeMix Mainnet" and you'll find it.

File: docs/CONFIG.md (hunk 6 above)
Comment (RensR) [reply docs/CONFIG.md]: Lets have BIX own that, not sure why this is like it is but they added it afaik

Comment (makramkd) [reply docs/CONFIG.md]: https://smartcontract-it.atlassian.net/browse/BCI-2530

File: docs/CONFIG.md (hunk 8 above)
Comment (makramkd) [reply docs/CONFIG.md]: https://smartcontract-it.atlassian.net/browse/BCI-2530

File: core/services/chainlink/application.go (hunk 2 above)
Comment (makramkd) [reply core/services/chainlink/application.go:244]: Will raise a ticket.

Comment (makramkd) [reply core/services/chainlink/application.go:244]: PR to fix in core: https://github.com/smartcontractkit/chainlink/pull/11603

File: core/services/keeper/delegate.go (hunk 11 above)
Comment (makramkd) [reply core/services/keeper/delegate.go:58]: Seems like these have been removed in core but we still need them on CCIP.

File: core/services/ocr2/plugins/ccip/internal/ccipdata/onramp_reader_test.go (hunk 4 above)
Comment (makramkd) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/onramp_reader_test.go]: Fixed.

File: integration-tests/ccip-tests/testconfig/global.go (hunk 9 above)
Comment (RensR) [reply integration-tests/ccip-tests/testconfig/global.go]: I mean all the imports have newline where they shouldn't have. Not really sure how that happens when these are the only changes made

File: core/services/chainlink/application.go (hunk 2 above)
Comment (makramkd) [reply core/services/chainlink/application.go:244]: PR above merged.

**Review by RensR — APPROVED** [review]

**Review by kalverra — APPROVED** [review]
